src/main/groovy/com/bmuschko/gradle/docker/internal/DefaultDockerConfigResolver.groovy

```diff
@@ -0,0 +1,66 @@
+package com.bmuschko.gradle.docker.internal
+
+import com.google.common.annotations.VisibleForTesting
+import groovy.transform.CompileStatic
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
+
+import javax.annotation.Nullable
+
+import static com.bmuschko.gradle.docker.internal.OsUtils.isWindows
+
+@CompileStatic
+class DefaultDockerConfigResolver {
```

Comment:
This should implement an interface so that it becomes mockable. Please remove the usable of `static` methods.

Reply:
Can you give a little bit snippet of how you see it?

Reply:
Something like:

```
interface DockerConfigResolver {
   String getDefaultDockerUrl()
    ...
}

class DefaultDockerConfigResolver implements DockerConfigResolver {
    @Override
    String getDefaultDockerUrl() {
        ...
    }
    ...
}
```

Reply:
Hmm, not very understand what it can bring to me, as I need to test the body of the `getDefaultDockerUrl()` method.

Reply:
It will help with testing where the class is being used, in the extension. I can touch that up after merging the PR.